Test failure in CI build 6031

The following test appears to have failed:

[#6031](https://circleci.com/gh/cockroachdb/cockroach/6031):

```
I0818 14:43:37.411387 471 raft/raft.go:468  group 2 100000001 received vote from 100000001 at term 6
I0818 14:43:37.411566 471 raft/raft.go:447  group 2 100000001 became leader at term 6
I0818 14:43:37.413123 471 storage/replica_command.go:999  range 2: new leader lease replica 1:1 00:00:00.000 +1.000s
I0818 14:43:37.413712 471 storage/replica_command.go:1061  initiating a split of range=2 ["db1"-"\xff\xff") at key "db2"
I0818 14:43:37.415942 471 client/db.go:451  failed Batch: node unavailable; try another peer
panic: error is unexpectedly set

goroutine 6251 [running]:
github.com/cockroachdb/cockroach/kv.(*LocalSender).Send(0xc2082d2100, 0x7f91c1e18fe8, 0xc208860ea0, 0x7f91c1e27120, 0xc2082ed180, 0x7f91c1e27228, 0xc208210090, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/kv/local_sender.go:168 +0x95d
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).sendOne(0xc20810e000, 0x7f91c1e18fe8, 0xc208860de0, 0x7f91c1e27120, 0xc2082ed180, 0x7f91c1e27228, 0xc208210090, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:427 +0xb01
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).Send(0xc20810e000, 0x7f91c1e18fe8, 0xc208860de0, 0x7f91c1e27120, 0xc2082ed180, 0x7f91c1e27228, 0xc208210090, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:317 +0x663
github.com/cockroachdb/cockroach/client.(*txnSender).Send(0xc20812ab60, 0x7f91c1e18fa8, 0xc208028b10, 0x7f91c1e27120, 0xc2082ed180, 0x7f91c1e27228, 0xc208210090, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/client/txn.go:52 +0x131
--
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:249 +0x4ea
github.com/cockroachdb/cockroach/util/stop.func·001()
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x8c
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:81 +0x14c
FAIL	github.com/cockroachdb/cockroach/storage	15.841s
=== RUN TestBatchBasics
I0818 14:43:44.775645 550 storage/engine/rocksdb.go:122  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.01s)
=== RUN TestBatchGet
I0818 14:43:44.782419 550 storage/engine/rocksdb.go:122  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.01s)
=== RUN TestBatchMerge
I0818 14:43:44.788588 550 storage/engine/rocksdb.go:122  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.01s)
=== RUN TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/kv/txn_coord_sender.go
+++ b/kv/txn_coord_sender.go
@@ -412,6 +412,7 @@ func (tc *TxnCoordSender) sendOne(ctx context.Context, call proto.Call) {
 				// TransactionAbortedError instead of this ambivalent error.
 				call.Reply.Header().SetGoError(util.Errorf(
 					"transaction is already committed or aborted"))
+				return
 			} else if len(args.Intents) == 0 {
 				// If there aren't any intents, then there's factually no
 				// transaction to end. Read-only txns have all of their state in

